dd-sdk-android-core/src/main/kotlin/com/datadog/android/v2/core/SdkInternalLogger.kt

```diff
@@ -75,33 +81,71 @@ internal class SdkInternalLogger(
     private fun logToUser(
         level: InternalLogger.Level,
         message: String,
-        error: Throwable?
+        error: Throwable?,
+        onlyOnce: Boolean
     ) {
-        devLogger.log(
-            level.toLogLevel(),
-            message.withSdkName(),
-            error
+        sendToLogHandler(
+            userLogger,
+            level,
+            message,
+            error,
+            onlyOnce,
+            onlyOnceUserMessages
         )
     }
 
     private fun logToMaintainer(
         level: InternalLogger.Level,
         message: String,
-        error: Throwable?
+        error: Throwable?,
+        onlyOnce: Boolean
     ) {
-        sdkLogger?.log(
-            level.toLogLevel(),
-            message.withSdkName(),
-            error
-        )
+        maintainerLogger?.let {
+            sendToLogHandler(
+                it,
+                level,
+                message,
+                error,
+                onlyOnce,
+                onlyOnceMaintainerMessages
+            )
+        }
+    }
+
+    private fun sendToLogHandler(
+        handler: LogcatLogHandler,
+        level: InternalLogger.Level,
+        message: String,
+        error: Throwable?,
+        onlyOnce: Boolean,
+        knownSingleMessages: MutableList<String>
+    ) {
+        if (onlyOnce) {
+            if (knownSingleMessages.contains(message)) {
+                // drop the message… wait should we log that we dropped it?
+                return
+            } else {
+                knownSingleMessages.add(message)
```

Comment:
I think it may be useful to add message attributed to the particular SDK instance. Because in the situation when 2 SDK instances are logging the same message, only message associated with first instance will be shown, masking the message/issue with the second instance as well.

Reply:
The Internal logger is already tied to a specific instance, meaning that message seen in one instance won't impact messages seen in another instance, or in the UNBOUND logger

Reply:
The Internal logger is already tied to a specific instance, meaning that message seen in one instance won't impact messages seen in another instance, or in the UNBOUND logger

Reply:
true, I missed that.